packages/test/test-end-to-end-tests/src/test/gcTests/gcDriverWrappers.ts

```diff
@@ -0,0 +1,74 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ITelemetryBaseLogger } from "@fluidframework/common-definitions";
+import {
+    IDocumentService,
+    IDocumentServiceFactory,
+    IDocumentStorageService,
+    IResolvedUrl,
+    ISummaryContext,
+} from "@fluidframework/driver-definitions";
+import { ISummaryTree } from "@fluidframework/protocol-definitions";
+
+/**
+ * Wraps the given IDocumentStorageService to override the `uploadSummaryWithContext` method. It calls the
+ * `uploadSummaryCb` whenever a summary is uploaded by the client.
+ */
+export function wrapDocumentStorageService(
```

Comment:
I think for tests this is fine.. and really I can't think of any reason not to do this except type safety.

If you have access to the driver-utils package, you could extend DocumentStorageServiceProxy instead.